src/sinks/prometheus/remote_write.rs

```diff
@@ -287,6 +365,31 @@ fn snap_block(data: Bytes) -> Vec<u8> {
         .expect("Out of memory")
 }
 
+pub async fn sign_request(
```

Comment:
sign_request is the same implementation as [in elasticsearch](https://github.com/vectordotdev/vector/blob/9b73df354e94385acc76a859c16fafea5c912e7c/src/sinks/elasticsearch/common.rs#L168-L191), it may be better to consolidate in one place

Reply:
Agreed. That could get moved up into `src/aws` and shared.